src/builder/create_components.rs

```diff
@@ -17,19 +16,20 @@ pub enum CreateActionRow<'a> {
     InputText(CreateInputText<'a>),
 }
 
-impl<'a> serde::Serialize for CreateActionRow<'a> {
+impl serde::Serialize for CreateActionRow<'_> {
     fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-        use serde::ser::Error as _;
-
-        json!({
-            "type": 1,
-            "components": match self {
-                Self::Buttons(x) => serde_json::to_value(x).map_err(S::Error::custom)?,
-                Self::SelectMenu(x) => serde_json::to_value(vec![x]).map_err(S::Error::custom)?,
-                Self::InputText(x) => serde_json::to_value(vec![x]).map_err(S::Error::custom)?,
-            }
-        })
-        .serialize(serializer)
+        use serde::ser::SerializeMap as _;
+
+        let mut map = serializer.serialize_map(Some(2))?;
+        map.serialize_entry("type", &1_u8)?;
+
+        match self {
+            CreateActionRow::Buttons(buttons) => map.serialize_entry("components", buttons)?,
+            CreateActionRow::SelectMenu(select) => map.serialize_entry("components", select)?,
+            CreateActionRow::InputText(input) => map.serialize_entry("components", input)?,
```

Comment:
These two should serialise as an array.
```suggestion
            CreateActionRow::SelectMenu(select) => map.serialize_entry("components", [select])?,
            CreateActionRow::InputText(input) => map.serialize_entry("components", [input])?,
```

Reply:
Fixed